src/hivpy/sex_behaviour_data.py

```diff
@@ -30,14 +34,23 @@ def _norm_probs(self, prob_dict: dict):
             for key, data in prob_dict.items()
         }
 
+    def _select_matrix(self, matrix_list):
+        return matrix_list[np.random.choice(len(matrix_list))]
```

Comment:
This would also work and appear more direct, unless I'm missing a dimension or something?
```suggestion
        return np.random.choice(matrix_list)
```

Reply:
The list of matrices gets loaded into a 3D array (this is done by py-yaml), but np.random.choice will only accept 1D arrays (it does not interpret this as an array of 2D arrays) e.g.:
`>>> x = np.array([[[1.0,0.0,0.0],[0.0,1.0,0.0],[0.0,0.0,1.0]],
...               [[1.0,1.0,1.0],[0.0,0.0,0.0],[0.0,0.0,0.0]],
...               [[0.0,0.0,0.0],[1.0,1.0,1.0],[0.0,0.0,0.0]]])
>>> np.random.choice(x)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "mtrand.pyx", line 911, in numpy.random.mtrand.RandomState.choice
ValueError: a must be 1-dimensional`

So this is kind of a fudge to get around that without altering the yaml import or moving the data into an intermediate structure.

Reply:
Okay, makes sense. I thought this was loaded as a list, not a 3D numpy array. The `random.choice` function from the standard library would work but it may be slower, and we'd have to keep track of multiple RNGs then.